src/components/Offices/Finances/index.js

```diff
@@ -33,37 +40,59 @@ const ACCOUNTS = [
 export default function Finances() {
   const dispatch = useDispatch();
   const balances = useSelector(officesSelectors.selectorBalances);
+  const pallets = useSelector(officesSelectors.selectorPallets);
+  const [accountsAddresses, setAccountsAddresses] = useState([]);
+
+  useEffect(() => {
+    dispatch(officesActions.getPalletIds.call());
+  }, [dispatch]);
 
   useEffect(() => {
-    dispatch(officesActions.getBalances.call(ACCOUNTS.map(a => a.address)));
-  }, [])
+    if (pallets) {
+      const palletIdsToFetchBalancesFor = [
+        ...DEFAULT_ACCOUNTS,
+        ...DYNAMIC_ACCOUNTS.map((account) => pallets.find((e) => e.palletName === account.codeName)),
+      ];
+
+      const addressesToFetchBalancesFor = palletIdsToFetchBalancesFor.map(
+        ({ palletId, address }) => (palletId ? palletIdToAddress(palletId) : address),
+      );
+      setAccountsAddresses(addressesToFetchBalancesFor);
+      dispatch(officesActions.getBalances.call(addressesToFetchBalancesFor));
+    }
+  }, [dispatch, pallets]);
+
+  if (!pallets || !balances) return 'Loading...';
 
   return (
     <Table
       columns={[
         {
-          Header: "Name",
-          accessor: "name",
+          Header: 'Name',
+          accessor: 'name',
         },
         {
-          Header: "Address",
-          accessor: "address"
+          Header: 'Address',
+          accessor: 'address',
         },
         {
-          Header: "LLM Balance",
-          accessor: "llm"
+          Header: 'LLM Balance',
+          accessor: 'llm',
         },
         {
-          Header: "LLD Balance",
-          accessor: "lld"
+          Header: 'LLD Balance',
+          accessor: 'lld',
         },
       ]}
-      data={ACCOUNTS.map(a => {
-        return Object.assign(a, {
-          llm: formatMerits(balances.LLM[a.address] ?? 0) + ' LLM',
-          lld: formatDollars(balances.LLD[a.address] ?? 0) + ' LLD',
-        })
+      data={[...DEFAULT_ACCOUNTS, ...DYNAMIC_ACCOUNTS].map((a, key) => {
```

Comment:
I don't like that we need to rely on array indices here. We're already building this array in second `useEffect`. Instead of storing just plain addresses in state, store array of objects with both display name and address.